### DISCUSSION TITLE
More performant

### DISCUSSION THREAD

**Discussion by jkenzer** [jamessimone/apex-rollup#147] [Q&A] [answered]
I'm trying to roll up a date field to get the most recent date of all the child records. Specifically, it's bills (child) associated with a vendor (parent - account object). So I'm looking to store the last bill date on the accounts object.

My question: Is it more performant to do a rollup operation of Max on a date field or a Last on a date field? The reason I'm asking is I'm getting some CPU timeout errors when trying to do a full recalc to do the initial populating of this field.

I'm currently using Last with the order by. Does the ASC vs DESC get set automatically based on the First vs Last?

**Comment by jkenzer** [comment]
I created a new Rollup Control and set it to Batchable instead of Queuable. Seems to be working better. Does this make sense to do for the initial populating and then switch it back when running in "maintenance mode"?

**Reply by jkenzer** [reply]
This didn't actually work. I didn't get the CPU timeouts, but only about 20% of the rollups got set.

**Reply by jamessimone** [reply]
Hmmm. I am working on some optimizations at the moment (in regards to #146, but should be applicable here), however the framework _should_ be automatically batching above a certain size limit even if the default `Should Run As` type is set to Queueable. With that being said - I would expect `LAST` to be more performant for your use-case.

**Reply by jkenzer** [reply]
Okay, I went the old fashion route and did an update on all the child records to let the trigger handler handle it. That worked fine.

**Reply by jamessimone** [reply]
Nice. Playing the game the way it was meant to be played. Apologies on the full recalc stuff, I will continue to work on hunting down what's going wrong there.

**Comment by jamessimone** [comment, marked as answer]
My suggested answer (going through Discussions for the first time in a while) - 

There should be no discernable performance difference between a MAX of a date field versus using LAST with an Order By on the same field. While the underlying mechanism producing the result between the two differs, the end result in terms of time should be nigh-indistinguishable.
